client-spark/spark2/src/main/java/org/apache/spark/shuffle/DelegationRssShuffleManager.java

```diff
@@ -66,12 +66,23 @@ public DelegationRssShuffleManager(SparkConf sparkConf, boolean isDriver) throws
   private ShuffleManager createShuffleManagerInDriver() throws RssException {
     ShuffleManager shuffleManager;
 
-    boolean canAccess = tryAccessCluster();
+    RssAccessClusterResponse response = tryAccessCluster();
+    boolean canAccess = false;
+    if (response != null && response.getStatusCode() == ResponseStatusCode.SUCCESS) {
+      canAccess = true;
+    }
+
     if (canAccess) {
       try {
+        if (!sparkConf.contains(RssClientConfig.RSS_STORAGE_TYPE)) {
+          sparkConf.set(RssClientConfig.RSS_STORAGE_TYPE, response.getStorageType());
+          sparkConf.set(RssClientConfig.RSS_BASE_PATH, response.getStoragePath());
+        }
         shuffleManager = new RssShuffleManager(sparkConf, true);
         sparkConf.set(RssClientConfig.RSS_ENABLED, "true");
+        sparkConf.set("spark.shuffle.manager", Constants.RSS_SHUFFLE_MANAGER_NAME);
```

Comment:
I think we shouldn't set `shuffle.manager`.

Reply:
This is kind of tricky will remove it.